phpBB/phpbb/plupload/plupload.php

```diff
@@ -276,22 +276,47 @@ public function generate_resize_string()
 	}
 
 	/**
-	* Checks various php.ini values and the maximum file size to determine
-	* the maximum size chunks a file can be split up into for upload
+	* Checks various php.ini values to determine the maximum chunk
+	* size a file should be split into for upload.
+	*
+	* The intention is to calculate a value which reflects whatever
+	* the most restrictive limit is set to.  And to then set the chunk
+	* size to half that value, to ensure any required transfer overhead
+	* and POST data remains well within the limit.  Or, if all of the
+	* limits are set to unlimited, the chunk size will also be unlimited.
 	*
 	* @return int
 	*/
 	public function get_chunk_size()
 	{
-		$max = min(
-			$this->php_ini->getBytes('upload_max_filesize'),
-			$this->php_ini->getBytes('post_max_size'),
-			max(1, $this->php_ini->getBytes('memory_limit')),
-			$this->config['max_filesize']
-		);
+		$max = 0;
+
+		// unlimited is -1 for memory_limit. 0 should be an invalid configuration.
+		$limit_memory = $this->php_ini->getBytes('memory_limit');
+
+		if ($limit_memory > 0)
+		{
+			$max = $limit_memory;
+		}
+
+		// For all remaining limits, 0 means "unlimited".
+
+		$limit_upload = $this->php_ini->getBytes('upload_max_filesize');
+
+		if ($limit_upload > 0)
+		{
+			$max = min($limit_upload, $max ? $max : $limit_upload);
```

Comment:
Ternary is better to be surrounded to avoid any issues.

`$max = min($limit_upload, ($max ? $max : $limit_upload));`